lib/dao.js

```diff
@@ -2638,8 +2638,12 @@ DataAccessObject.replaceById = function(id, data, options, cb) {
   assert(typeof cb === 'function', 'The cb argument must be a function');
 
   var connector = this.getConnector();
-  assert(typeof connector.replaceById === 'function',
-          'replaceById() must be implemented by the connector');
+
+  if (typeof connector.replaceById !== 'function') {
+    var err =  new Error('there is no support for replaceById() in the connector.' +
+      'Connector needs to add support for `replaceById()`');
+    return cb(err);
+  }
```

Comment:
I don't think this is a breaking change. We still report an error, but don't crash the app.

Let's consider a user using CloudantDB connector (that does not implement replaceById yet). When this user upgrades to the newer version of LoopBack, they will suddenly get few more REST endpoints for replaceById and replaceOrCreate. These endpoints are not used by the app at all, but they present a vulnerability because calling them will crash the application via `assert`.

By reporting the error back to the caller, instead of crashing the server, we make the upgrade process easier.